eyecite/resolve.py

```diff
@@ -0,0 +1,177 @@
+from collections import defaultdict
+from typing import Any, Callable, Dict, KeysView, List, Optional
+
+from eyecite.models import (
+    CitationBase,
+    FullCaseCitation,
+    IdCitation,
+    NonopinionCitation,
+    Resource,
+    ShortCaseCitation,
+    SupraCitation,
+)
+from eyecite.utils import strip_punct
+
+
+def _resolve_fullcase_citation(citation: FullCaseCitation) -> Resource:
```

Comment:
This might want to be named something without an underscore, since it's probably a best practice to import it and call it on unmatched citations when overriding it? Not sure about the other three, but this one feels like part of the API.

Reply:
I'm not sure I follow Jack? Does this make sense to you, Matt? If so, I'm happy to just see a change or not.

Reply:
Sorry, to be less opaque, I'm referring to the PEP8 rule that variable names starting with underscore are part of the internal interface of a module and are allowed to change without warning. I don't think this method qualifies, because people should be encouraged to import it and call it from user code in the documentation. So I'd suggesting naming it something that is (a) public and (b) future-proof when we add non-case citation extraction. How about changing `_resolve_fullcase_citation` to `resolve_full_citation`?

Why public and documented? Because the reason we return a dict from `resolve_citations` is to help people who override this function, and when they do that they ought to call the base function from the override. So something like this should be encouraged in the docs:

```
from eyecite.resolve import resolve_full_citation

def my_resolve(full_cite):
    # special handling for resolution of known cases in our database
    resource = MyOpinion.objects.filter(citation__reporter=...).first()
    if resource:
        return resource
    # allow normal clustering of other citations
    return resolve_full_citation(full_cite)

resolve_citations(citations, resolve_full_citation=resolve_full_citation)
{
    <MyOpinion object>: [<full_cite>, <short_cite>, <id_cite>],
    <Resource object>: [<full cite>, <short cite>],
}
```

Reply:
This seems logical to me, thanks for explaining more.

Reply:
Understood the underscore part, but didn't think calling the function at the end of the override would be useful or standard, but happy to be wrong. Resolving.